nix/tabry-hm-module.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+# This file contains a Home Manager module that installs tabry
+# and sets up tabry configuration files to be used by tabry
+#
+let
+
+  cfg = config.programs.tabry;
+
+  tabry = pkgs.callPackage ../default.nix {};
+  tabryLang = pkgs.callPackage ./tabry-lang.nix { inherit tabry; };
+
+  # converts /nix/store/.../foo.tabry to "foo"
+  commandNameFromTabryFilename = fileName: 
+    (builtins.replaceStrings [".tabry"] [""] (builtins.baseNameOf fileName));
+
+  mkInitFish = fileName: let
+    commandName = commandNameFromTabryFilename fileName;
+  in ''
+    tabry_completion_init ${commandName}
+  '';
+
+  compileTabryFiles = map tabryLang.compileTabryFile;
+
+in {
+
+  options.programs.tabry = {
+    enable = lib.mkEnableOption "tabry, a tab completion library";
+    enableFishIntegration = lib.mkEnableOption "enables fish completions";
+    enableBashIntegration = lib.mkEnableOption "enables bash completions";
+    tabryFiles = lib.mkOption {
+      type = with lib.types; listOf path;
+      default = [];
+      description = ''
+        *.tabry files to be compiled to completion json
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable (
+    let
+      tabryImportsPath = builtins.concatStringsSep ":" (compileTabryFiles cfg.tabryFiles);
+    in {
+      home.packages = [tabry];
+
+      # for each file, compile it to json
+      # then add the dir to $TABRY_IMPORTS_PATH
+
+      programs.fish.shellInit = lib.mkIf cfg.enableFishIntegration (
+        ''
+          set -x TABRY_IMPORT_PATH "${tabryImportsPath}:$TABRY_IMPORTS_PATH"
```

Comment:
The second one on this line should also be $TABRY_IMPORT_PATH, right? (And in the bash.initExtra below, and the comment above)